dvc/fs/azure.py

```diff
@@ -69,8 +69,9 @@ def _prepare_credentials(self, config):
         login_info["client_id"] = config.get("client_id")
         login_info["client_secret"] = config.get("client_secret")
 
-        if not any(login_info.values()):
-            login_info["credentials"] = DefaultAzureCredential()
+        non_cred_auth_methods = ["connection_string", "client_secret", "account_key", "sas_token"]
+        if not any([login_info.get(a, None) is not None for a in non_cred_auth_methods]):
```

Comment:
If what you want to exclude is only `account_name`, I'd go with
```suggestion
        if not any(value for key, value in login_info.items() if key != 'account_name'):
```